src/iOS/toga_iOS/widgets/optioncontainer.py

```diff
@@ -0,0 +1,134 @@
+    #                                                  tbc,           # type: UITabBarController
+    #                                                  vc,            # type: UIViewController
+    #                                                  ) -> bool:
+    #     """
+    #     Event callback before tab has transitioned to new tab.
+    #     """
+    #
+    #     # print(f'TogaTabBarController.tabBarController_shouldSelectViewController_(tbc,vc)')
+    #
+    #     result = True
+    #     return result
+
+    @objc_method
+    def tabBarController_didSelectViewController_(self,
+                                                  tbc,              # type: UITabBarController
+                                                  vc,               # type: UIViewController
+                                                  ) -> None:
+        """
+        Event callback after tab has finished transitioning to new tab.
+        """
+
+        # call the user on_select callback method (if provided)
+        if self.interface.on_select:
+            index = tbc.selectedIndex
+            option = tbc.interface.content[index]
+            self.interface.on_select(self.interface, option=option, index=index)
+
+        # required to redraw new tabs correctly (possible Toga bug?)
+        self.interface.refresh()
+
+
+class OptionContainer(Widget):
+    """
+    OptionContainer implementation for the iOS platform.
+    """
+
+    def create(self):
+        """
+        Create native widget - iOS UITabController
+        """
+
+        self.controller = TogaTabBarController.alloc().init()
+        self.controller.interface = self.interface
+
+        self.native = self.controller.view
+
+        self.controller.delegate = self.controller
+        self.native.delegate = self.controller
+
+        # a list of view controllers for each tab (see `add_content()`)
+        self.view_controllers = []
+
+        # self.native.translatesAutoresizingMaskIntoConstraints = False
+
+        # Add the layout constraints
+        self.add_constraints()
+
+    def add_content(self, label, widget):
+        """ Adds a new option to the option container.
+
+        Args:
+            label (str): The label for the option container
+            widget: The widget or widget tree that belongs to the label.
+        """
+
+        widget.viewport = iOSViewport(widget.native)
+
+        for child in widget.interface.children:
+            child._impl.container = widget
+
+        # tag = len(self.interface.content) - 1
+        tab_bar_item = UITabBarItem.alloc().init()
+        tab_bar_item.title = label
+
+        # Turn the autoresizing mask on the widget widget
+        # into constraints. This makes the widget fill the
+        # available space inside the OptionContainer.
+        widget.native.translatesAutoresizingMaskIntoConstraints = True
+
+        # create view controller for the view and associate tab bar item
+        vc = UIViewController.alloc().init()
```

Comment:
This will definitely work for the "simple box" case; however, some widgets are their own controllers. For example,  if you had a complex UI with tab navigation, but one fo the tabs usedb UINavigationController for some heirarchial content, the widget being added as a tab would have it's own controller. 

set_content on Window already had some support for this. I'm not 100% happy with that approach - I think there might be a need to introduce the concept of a "top level container" in Toga (since that's essentially what the ViewController is managing here).

Reply:
I'm not fully across this.  Is it as simple as "if widget is a controller, then no need to create a UIViewController" ?

Can we check with "if isinstance(widget, UIViewController)" ?

Reply:
The widget won't *ever* be a UIViewController - it's a Toga widget. What you're looking for is whether the *widget* is acting as a controller.

Have you taken a look at set_content on Window? The immediate workaround logic you need is in that method - it's the lines that reference UIViewController.